rnote-ui/src/workspacebrowser/filerow/action/duplicate/mod.rs

```diff
@@ -0,0 +1,164 @@
+use std::ffi::{OsStr, OsString};
+use std::path::PathBuf;
+
+use fs_extra::dir::{CopyOptions, TransitProcessResult};
+use fs_extra::{copy_items_with_progress, TransitProcess};
+use gtk4::prelude::FileExt;
+use gtk4::{gio, glib, glib::clone};
+use regex::Regex;
+
+use crate::workspacebrowser::FileRow;
+use crate::RnoteAppWindow;
+
+///                                 - Look for `.dup` pattern
+///                                 |   - Look for `.dup1`/`.dup123`/`.dup1234`/...
+///                                 |   |        - Look for the text after the `.dup<num>` part
+///                                 |   |       |       - At the end of the word (here: file-path)
+///                                 |  \d*      |       $
+///                                 |           |
+///                               \.dup   (?P<rest>(.*))
```

Comment:
Thanks for that explanation. Super helpful for maintaining the code in the future.